DocsCode/NavigationTutorial/NavigationTutorialViewController.swift

```diff
@@ -0,0 +1,122 @@
+        // #-end-code-snippet: navigation init-map-swift
+        
+        // #-code-snippet: navigation user-location-swift
+        // Allow the map view to display the user's location
+        mapView.showsUserLocation = true
+        // #-end-code-snippet: navigation user-location-swift
+        
+        // #-code-snippet: navigation gesture-recognizer-swift
+        // Add a gesture recognizer to the map view
+        let setDestination = UILongPressGestureRecognizer(target: self, action: #selector(didLongPress(_:)))
+        mapView.addGestureRecognizer(setDestination)
+        // #-end-code-snippet: navigation gesture-recognizer-swift
+    }
+    
+    // #-code-snippet: navigation allow-callouts-swift
+    // Implement the delegate method that allows annotations to show callouts when tapped
+    func mapView(_ mapView: MGLMapView, annotationCanShowCallout annotation: MGLAnnotation) -> Bool {
+        return true
+    }
+    // #-end-code-snippet: navigation allow-callouts-swift
+    
+    // #-code-snippet: navigation long-press-swift
+    @objc func didLongPress(_ sender: UILongPressGestureRecognizer) {
+        guard sender.state == .began else { return }
+        
+        // Converts point where user did a long press to map coordinates
+        let point = sender.location(in: mapView)
+        let coordinate = mapView.convert(point, toCoordinateFrom: mapView)
+        
+        // Create a basic point annotation and add it to the map
+        let annotation = MGLPointAnnotation()
+        annotation.coordinate = coordinate
+        annotation.title = "Start navigation"
+        mapView.addAnnotation(annotation)
+        
+        // Calcuate the route from the user's location to the set destination
+        calculateRoute(from: (mapView.userLocation!.coordinate), to: annotation.coordinate) { [unowned self] (route, error) in
+            if error != nil {
+                // Print an error message
+                print("Error calculating route")
+            }
+        }
+    }
+    // #-end-code-snippet: navigation long-press-swift
+    
+    // #-code-snippet: navigation calculate-route-swift
+    // Calculate route to be used for navigation
+    func calculateRoute(from origin: CLLocationCoordinate2D,
+                        to destination: CLLocationCoordinate2D,
+                        completion: @escaping (Route?, Error?) -> ()) {
+        
+        let origin = Waypoint(coordinate: origin, name: "Start")
+        let destination = Waypoint(coordinate: destination, name: "Finish")
+        
+        // Specify that the route is intented for automobiles avoiding traffic
+        let options = NavigationRouteOptions(waypoints: [origin, destination], profileIdentifier: .automobileAvoidingTraffic)
+        
+        // Generate the route object and draw it on the map
+        _ = Directions.shared.calculate(options) { (waypoints, routes, error) in
+            guard let route = routes?.first else { return }
```

Comment:
You could skip creating this `route` variable and directly assign to `self.directionsRoute` here.